nntrainer/tensor/var_grad.h

```diff
@@ -176,7 +175,16 @@ class Var_Grad {
     resetGradient();
   }
 
-protected:
+  void setBatchSize(unsigned int batch) {
+    dim.batch(batch);
+    /** @note This will shape when changing batch size with initialized
+     * variables */
+    if (!var->uninitialized())
+      var->reshape(dim);
```

Comment:
I wonder if this is not throwing as dimension change is not allowed if datalen changes.

Reply:
this is intended to throw.
Later there will be a reset which will reset there variables (ideally this memory will be freed after training/inference is done). Then batch size could be changed.

Reply:
Acked